flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/catalog/UnknownCatalogTest.java

```diff
@@ -55,6 +56,40 @@ public class UnknownCatalogTest {
     public static final ResolvedSchema EXPECTED_SCHEMA =
             ResolvedSchema.of(Column.physical("i", INT()), Column.physical("s", STRING()));
 
+    public static final ResolvedSchema CURRENT_TIMESTAMP_EXPECTED_SCHEMA =
+            ResolvedSchema.of(Column.physical("CURRENT_TIMESTAMP", TIMESTAMP_LTZ(3).notNull()));
+
+    @Test
+    public void testUnsetCatalogWithSelectCurrentTimestamp() throws Exception {
+        TableEnvironment tEnv = TableEnvironment.create(ENVIRONMENT_SETTINGS);
+
+        tEnv.useCatalog(null);
+        Table table = tEnv.sqlQuery("SELECT CURRENT_TIMESTAMP;");
+
+        assertThat(table.getResolvedSchema()).isEqualTo(CURRENT_TIMESTAMP_EXPECTED_SCHEMA);
+    }
+
+    @Test
+    public void testSetCatalogUnsetDatabaseWithSelectCurrentTimestamp() throws Exception {
+        TableEnvironment tEnv = TableEnvironment.create(ENVIRONMENT_SETTINGS);
+
+        tEnv.useCatalog(BUILTIN_CATALOG);
+        tEnv.useDatabase(null);
+        Table table = tEnv.sqlQuery("SELECT CURRENT_TIMESTAMP;");
```

Comment:
One last comment: please drop the semicolon, it has only a meaning in scripts, not code.